packages/ti_anomali/data_stream/intelligence/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,552 @@
+        ransomware_ipv6: ipv6-addr
+        ransomware_url: url
+        rootkit_hash: file
+        scan_ip: ipv4-addr
+        scan_ipv6: ipv6-addr
+        sinkhole_domain: domain-name
+        sinkhole_ip: ipv4-addr
+        sinkhole_ipv6: ipv6-addr
+        social_media_url: url
+        spam_domain: domain-name
+        spam_email: email-addr
+        spam_email_subject: email
+        spam_ip: ipv4-addr
+        spam_ipv6: ipv6-addr
+        spam_mta: null
+        spam_url: url
+        speedtest_url: url
+        ssh_ip: ipv4-addr
+        ssh_ipv6: ipv6-addr
+        ssl_cert_serial_number: null
+        suppress: suppress
+        suspicious_domain: domain-name
+        suspicious_email: email-addr
+        suspicious_email_subject: email
+        suspicious_ip: ipv4-addr
+        suspicious_reg_email: email-addr
+        suspicious_url: url
+        tor_ip: ipv4-addr
+        tor_ipv6: ipv6-addr
+        torrent_tracker_url: url
+        trojan_domain: domain-name
+        trojan_hash: file
+        trojan_ip: ipv4-addr
+        trojan_ipv6: ipv6-addr
+        trojan_url: url
+        vpn_domain: domain-name
+        vps_ip: ipv4-addr
+        vps_ipv6: ipv6-addr
+        whois_bulk_reg_email: email-addr
+        whois_privacy_domain: domain-name
+        whois_privacy_email: email-addr
+      source: >
+        if (params.containsKey(ctx.json.itype) && params[ctx.json.itype] != null) {
+          if (ctx.threat == null) {
+            ctx.threat = [:];
+          }
+          if (ctx.threat.indicator == null) {
+            ctx.threat.indicator = [:];
+          }
+          ctx.threat.indicator.type = params[ctx.json.itype];
+        } else {
+          throw new IllegalArgumentException("indicator type not found from itype");
+        }
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Unable to determine STIX 2.0 indicator type from itype "{{{ json.itype }}}": {{{ _ingest.on_failure_message }}}'
```

I was wondering if this could be written easily to avoid the use of exceptions. It cannot, and this is likely to be infrequent.

---

Yeah, I try to avoid them. Here it would need to check and create `error`, and `error.message`, then append to it. I think it's okay as is. It's not silencing failures that could have been handled.